### PR TITLE
Basic rock and laser collisions

### PR DESCRIPTION
**Pull request by squid-ai**
Work in progress

-created a `SpaceCollision` class that will check for collisions and remove the laser and the rock
-need access to the list of laser and rock GameObjects

### CODE REVIEW

**Comment by squid-ai** [comment]
Ready for review! When the rock and lasers collide, they will disappear.

File: src/seagulls/examples/space_shooter/_space_collisions.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        self._ship = ship
+        self._asteroid_field = asteroid_field
+
+    def tick(self) -> None:
+        _remove_lasers = []
+        _remove_rocks = []
+
+        for laser in range(self._ship.get_number_of_lasers()):
+            for asteroid in range(self._asteroid_field.get_asteroid_field_size()):
+                if self._asteroid_field.get_rock_position_x(asteroid) <= \
```
Comment (Zeelot) [inline src/seagulls/examples/space_shooter/_space_collisions.py:26]: can you help me understand what these if statements are doing? what does `<=` do?

Comment (squid-ai) [reply src/seagulls/examples/space_shooter/_space_collisions.py:26]: I'm checking the boundaries of the `(x, y)` positions of the laser relative to the asteroid. 

On either side of the expression is the asteroid's boundary, so it's `x` and `x + x-width of the rock`. I then use the `<=` to check if the laser's x coordinate is sandwiched in between those two boundaries. 

The second expression checks that the laser's `y` is also within the `y` boundaries of the rock (although it should hit just the bottom edge).

Comment (Zeelot) [reply src/seagulls/examples/space_shooter/_space_collisions.py:26]: oh! I think I just had some trouble following the multi line statement :) can we refactor this to a function call and simplify the conditions?

Comment (squid-ai) [reply src/seagulls/examples/space_shooter/_space_collisions.py]: great idea! refactored the conditional statements into function calls (and updated a few other sections) - let me know if this is what you had in mind :)

**Review by Zeelot — APPROVED** [review]

**Pull request merged**